### PR TITLE
[sdk] Remove uses of UIWebView

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Otherwise Apple may reject our apps.

# How

Reimplemented `expo-print` functionality using `WKWebView`.

x-ref: https://github.com/expo/react-native/pull/17

# Test Plan

NCL Print screen looks ok.

### CODE REVIEW

File: packages/expo-print/ios/EXPrint/EXWKPDFRenderer.m [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    return;
+  }
+
+  _onRenderingFinished(error, nil, 0);
+}
+
+- (WKWebView *)createWebView
+{
+  WKWebViewConfiguration *configuration = [[WKWebViewConfiguration alloc] init];
+//  configuration.userContentController = [self userContentController];
```
Comment (tsapeta) [inline packages/expo-print/ios/EXPrint/EXWKPDFRenderer.m]:
```suggestion
```

File: packages/expo-print/ios/EXPrint/EXWKPDFRenderer.m [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  NSString *addMetaScript = @""
+  "var meta = document.createElement('meta');"
+  "meta.setAttribute('name', 'viewport');"
+  "meta.setAttribute('content', 'width=device-width');"
+  "meta.setAttribute('initial-scale', '1.0');"
+  "meta.setAttribute('maximum-scale', '1.0');"
+  "meta.setAttribute('minimum-scale', '1.0');"
+  "meta.setAttribute('user-scalable', 'no');"
+  "document.getElementsByTagName('head')[0].appendChild(meta);";
+  return [[WKUserScript alloc] initWithSource:addMetaScript injectionTime:WKUserScriptInjectionTimeAtDocumentEnd forMainFrameOnly:YES];
```
Comment (tsapeta) [inline packages/expo-print/ios/EXPrint/EXWKPDFRenderer.m]: Shouldn't we put this at the document beginning? so people would be able to override this behavior

**Review by tsapeta — APPROVED** [review]

File: packages/expo-print/ios/EXPrint/EXWKPDFRenderer.m (hunk 2 above)
Comment (sjchmiela) [reply packages/expo-print/ios/EXPrint/EXWKPDFRenderer.m]: Ah sorry, I forgot to remove this code altogether

**Review by ide — APPROVED** [review]
Approving so we can start versioning and cut the release branch but left some comments for requested changes.

File: packages/expo-print/ios/EXPrint/EXWKViewPDFRenderer.m [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  CGRect containerFrame = CGRectMake(window.bounds.size.width, 0, webView.frame.size.width, webView.frame.size.height);
+  [_containerView setFrame:containerFrame];
+  CGRect webViewFrame = CGRectMake(0, 0, webView.frame.size.width, webView.frame.size.height);
+  [_containerView addSubview:webView];
+  [webView setFrame:webViewFrame];
+  [webView.scrollView setClipsToBounds:NO];
+  [window addSubview:_containerView];
+
+  UM_WEAKIFY(self);
+  dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
```
[the remark this answers is not in the document]
Comment (sjchmiela) [reply packages/expo-print/ios/EXPrint/EXWKViewPDFRenderer.m]: I didn't find a better way to trigger `WKWebView`'s rendering. Without the delay the only things that are rendered are `div`s, no text.

File: packages/expo-print/ios/EXPrint/EXWKViewPDFRenderer.m [hunk 4]
```diff
@@ -0,0 +1,97 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import <EXPrint/EXWKViewPDFRenderer.h>
+#import <UMCore/UMDefines.h>
+
+@interface EXWKViewPDFRenderer ()
```
[the remark this answers is not in the document]
Comment (sjchmiela) [reply packages/expo-print/ios/EXPrint/EXWKViewPDFRenderer.m]: Added in header file

File: packages/expo-print/ios/EXPrint/EXWKViewPDFRenderer.m (hunk 3 above)
[the remark this answers is not in the document]
Comment (sjchmiela) [reply packages/expo-print/ios/EXPrint/EXWKViewPDFRenderer.m]: Added small explanatory comment

https://github.com/expo/expo/pull/5578/files#diff-b21043019b44ad078174d6a97c441b38R41-R42

**Pull request merged**